restapi/src/main/java/io/stargate/web/docsapi/service/util/DocsApiUtils.java

```diff
@@ -61,6 +62,37 @@ public static String convertArrayPath(String path) {
     }
   }
 
+  /**
+   * Converts any literal unicode points into their actual character. E.g. if the input string is
+   * literally abc\u002E123, this function returns abc.123 This allows a user to use unicode escape
+   * sequence in where filters when it would otherwise be ambiguous to use the corresponding
+   * character. Any path that already has square brackets in it will be left alone.
+   *
+   * @param path single filter or field path
+   * @return Converted to a string with no literal unicode code points.
+   */
+  public static String convertUnicodeCodePoints(String path) {
+    if (ARRAY_PATH_PATTERN.matcher(path).matches()) {
+      return path;
+    }
+
+    String[] segments = path.split("\\\\u");
+    for (int i = 1; i < segments.length; i++) {
+      String segment = segments[i];
+      if (segment.length() >= 4
+          && UNICODE_POINT_PATTERN.matcher(segment.substring(0, 4)).matches()) {
```

Comment:
IMHO, Unicode points are inconvenient from the end-user's perspective. I guess in many cases query parameter may be written by hand, so figuring out the right escape sequence is going to be tedious and error-prone.

Would `\` escapes be an option (like in java: `"\""`, in Docs API: `field1\.partTwo`, `\\` - literal `\` char)?

Reply:
@ashryanbeats : Do you have any guidelines wrt escaping methods for Docs API query parameters (this is to happen after URL decoding).

Reply:
Yea this is a good question...one of the benefits of allowing unicode is that you can put other types of characters as well, but that's not necessarily the most useful thing in the world...

I think there might be a way to accept frontslash escaping for periods, braces, and quotes, and maybe that would do the trick? I'm not married to using unicode escapes, I agree that it might be a bit unwieldy - but I also figured for Javascript and Java developers it would be familiar to them at the very least